website/lib/markdown/index.js

```diff
@@ -73,7 +73,10 @@ module.exports = {
       'patterns',
       'error',
     ];
-    const docsURLs = flatPageListJson.flat.map((page) => page.pageURL);
+    const docsURLs = flatPageListJson.flat
+      // since this list will used to generate the sitemap.xml file, we need to hide the `hidden` pages from the output
+      .filter((page) => !page.pageAttributes?.navigation?.hidden)
```

Comment:
@Dhaulagiri do you see any downside to excluding the "hidden" pages from the list passed to `prember`?

Reply:
No downsides that I can think of. I'm not sure it's working though as I still see the hidden pages in the preview's [sitemap](https://hds-website-git-new-docs-website-frontmatter-updates-hashicorp.vercel.app/sitemap.xml)

Reply:
@Dhaulagiri something strange is happening:

if I do a `console.log()` of the docsURLs (and its length) this is what I see in the console:
```
⠙ building... [Babel: ember-prism > applyPatches]unexpectedly found "! ~" when slicing source, but expected " "
unexpectedly found "! ~" when slicing source, but expected " "
unexpectedly found "! ~" when slicing source, but expected " "
⠹ building... [Babel: ember-prism > applyPatches]unexpectedly found "! ~" when slicing source, but expected " "
⠋ building... [ember-auto-import-webpack] 59 [
  'icons/library',
  'icons/usage-guidelines',
  'about/overview',
  'about/principles',
  'about/support',
  'about/contribution',
  'about/accessibility-statement',
  'getting-started/for-designers',
  'getting-started/for-engineers',
  'foundations/accessibility',
  'foundations/border',
  'foundations/colors',
  'foundations/elevation',
  'foundations/focus-ring',
  'foundations/tokens',
  'foundations/typography',
  'components/alert',
  'components/badge',
  'components/badge-count',
  'components/breadcrumb',
  'components/button',
  'components/button-set',
  'components/card',
  'components/dropdown',
  'components/flyout',
  'components/form/primitives',
  'components/form/checkbox',
  'components/form/radio',
  'components/form/radio-card',
  'components/form/select',
  'components/form/text-input',
  'components/form/textarea',
  'components/form/toggle',
  'components/icon-tile',
  'components/link/inline',
  'components/link/standalone',
  'components/modal',
  'components/stepper',
  'components/table',
  'components/tabs',
  'components/tag',
  'components/toast',
  'overrides/power-select',
  'utilities/disclosure',
  'utilities/dismiss-button',
  'utilities/interactive',
  'testing/components/badge',
  'testing/components/banner',
  'testing/components/copy-button',
  'testing/components/do-dont',
  'testing/components/layout',
  'testing/components/typography',
  'testing/markdown/basic-styling',
  'testing/markdown/basic-spacing',
  'testing/markdown/more-complex-code',
  'testing/markdown/more-complex-images',
  'testing/markdown/more-complex-lists',
  'testing/markdown/more-complex-tables',
  'testing/markdown/with-ember-components'
]
DEBUG: -------------------------------
DEBUG: Ember : 4.8.1
DEBUG: -------------------------------
pre-render / 200 OK
pre-render about 200 OK
pre-render foundations 200 OK
pre-render components 200 OK
pre-render patterns 200 OK
pre-render error 200 OK
pre-render icons/library 200 OK
pre-render icons/usage-guidelines 200 OK
pre-render about/overview 200 OK
pre-render about/principles 200 OK
pre-render about/support 200 OK
pre-render about/contribution 200 OK
pre-render about/accessibility-statement 200 OK
pre-render getting-started/for-designers 200 OK
pre-render getting-started/for-engineers 200 OK
pre-render foundations/accessibility 200 OK
pre-render foundations/border 200 OK
pre-render foundations/colors 200 OK
pre-render foundations/elevation 200 OK
pre-render foundations/focus-ring 200 OK
pre-render foundations/tokens 200 OK
pre-render foundations/typography 200 OK
pre-render components/alert 200 OK
pre-render components/badge 200 OK
pre-render components/badge-count 200 OK
pre-render components/breadcrumb 200 OK
pre-render components/button 200 OK
pre-render components/button-set 200 OK
pre-render components/card 200 OK
pre-render components/dropdown 200 OK
pre-render components/flyout 200 OK
pre-render components/form/primitives 200 OK
pre-render components/form/checkbox 200 OK
pre-render components/form/radio 200 OK
pre-render components/form/radio-card 200 OK
pre-render components/form/select 200 OK
pre-render components/form/text-input 200 OK
pre-render components/form/textarea 200 OK
pre-render components/form/toggle 200 OK
pre-render components/icon-tile 200 OK
pre-render components/link/inline 200 OK
pre-render components/link/standalone 200 OK
pre-render components/modal 200 OK
pre-render components/stepper 200 OK
pre-render components/table 200 OK
pre-render components/tabs 200 OK
pre-render components/tag 200 OK
pre-render components/toast 200 OK
pre-render overrides/power-select 200 OK
pre-render utilities/disclosure 200 OK
pre-render utilities/dismiss-button 200 OK
pre-render utilities/interactive 200 OK
pre-render testing/components/badge 200 OK
pre-render testing/components/banner 200 OK
pre-render testing/components/copy-button 200 OK
pre-render testing/components/do-dont 200 OK
pre-render testing/components/layout 200 OK
pre-render testing/components/typography 200 OK
pre-render testing/markdown/basic-styling 200 OK
pre-render testing/markdown/basic-spacing 200 OK
pre-render testing/markdown/more-complex-code 200 OK
pre-render testing/markdown/more-complex-images 200 OK
pre-render testing/markdown/more-complex-lists 200 OK
pre-render testing/markdown/more-complex-tables 200 OK
pre-render testing/markdown/with-ember-components 200 OK

Build successful (37118ms) – Serving on http://localhost:4201/
```   
so clearly, the filter works from the JS perspective.

From what I understand from [the documentation](https://github.com/ef4/prember#using-prember-from-an-addon) this should work (and the pre-render list tells me it does).

Now, the sitemap _should_ be generated accordingly, looking at [the documentation](https://github.com/shipshapecode/prember-sitemap-generator#usage) but if I try to test it locally, it doesn't work (`/sitemap.xml` goes to the error page). 

My suspect is that it's the production build that is not taking into account the list of files? If I run `yarn build` locally, this is what I see:
```
~/src/hashicorp/design-system/website (new-docs-website/frontmatter-updates) $ yarn build
Environment: production
⠏ building... [Babel: @glimmer/component > applyPatches]unexpectedly found "{{! template-lint-d" when slicing source, but expected "@destinationElement"
unexpectedly found "{{! template-l" when slicing source, but expected "@renderInPlace"
⠹ building... [Babel: ember-power-select > applyPatches]unexpectedly found "! ~" when slicing source, but expected " "
⠸ building... [Babel: ember-power-select > applyPatches]unexpectedly found "! ~" when slicing source, but expected " "
unexpectedly found "! ~" when slicing source, but expected " "
unexpectedly found "! ~" when slicing source, but expected " "
⠼ building... [ember-auto-import-webpack]AAAA 59 [
  'icons/library',
  'icons/usage-guidelines',
  'about/overview',
  'about/principles',
  'about/support',
  'about/contribution',
  'about/accessibility-statement',
  'getting-started/for-designers',
  'getting-started/for-engineers',
  'foundations/accessibility',
  'foundations/border',
  'foundations/colors',
  'foundations/elevation',
  'foundations/focus-ring',
  'foundations/tokens',
  'foundations/typography',
  'components/alert',
  'components/badge',
  'components/badge-count',
  'components/breadcrumb',
  'components/button',
  'components/button-set',
  'components/card',
  'components/dropdown',
  'components/flyout',
  'components/form/primitives',
  'components/form/checkbox',
  'components/form/radio',
  'components/form/radio-card',
  'components/form/select',
  'components/form/text-input',
  'components/form/textarea',
  'components/form/toggle',
  'components/icon-tile',
  'components/link/inline',
  'components/link/standalone',
  'components/modal',
  'components/stepper',
  'components/table',
  'components/tabs',
  'components/tag',
  'components/toast',
  'overrides/power-select',
  'utilities/disclosure',
  'utilities/dismiss-button',
  'utilities/interactive',
  'testing/components/badge',
  'testing/components/banner',
  'testing/components/copy-button',
  'testing/components/do-dont',
  'testing/components/layout',
  'testing/components/typography',
  'testing/markdown/basic-styling',
  'testing/markdown/basic-spacing',
  'testing/markdown/more-complex-code',
  'testing/markdown/more-complex-images',
  'testing/markdown/more-complex-lists',
  'testing/markdown/more-complex-tables',
  'testing/markdown/with-ember-components'
]
pre-render / 200 OK
pre-render about 200 OK
pre-render foundations 200 OK
pre-render components 200 OK
pre-render patterns 200 OK
pre-render error 200 OK
pre-render icons/library 200 OK
pre-render icons/usage-guidelines 200 OK
pre-render about/overview 200 OK
pre-render about/principles 200 OK
pre-render about/support 200 OK
pre-render about/contribution 200 OK
pre-render about/accessibility-statement 200 OK
pre-render getting-started/for-designers 200 OK
pre-render getting-started/for-engineers 200 OK
pre-render foundations/accessibility 200 OK
pre-render foundations/border 200 OK
pre-render foundations/colors 200 OK
pre-render foundations/elevation 200 OK
pre-render foundations/focus-ring 200 OK
pre-render foundations/tokens 200 OK
pre-render foundations/typography 200 OK
pre-render components/alert 200 OK
pre-render components/badge 200 OK
pre-render components/badge-count 200 OK
pre-render components/breadcrumb 200 OK
pre-render components/button 200 OK
pre-render components/button-set 200 OK
pre-render components/card 200 OK
pre-render components/dropdown 200 OK
pre-render components/flyout 200 OK
pre-render components/form/primitives 200 OK
pre-render components/form/checkbox 200 OK
pre-render components/form/radio 200 OK
pre-render components/form/radio-card 200 OK
pre-render components/form/select 200 OK
pre-render components/form/text-input 200 OK
pre-render components/form/textarea 200 OK
pre-render components/form/toggle 200 OK
pre-render components/icon-tile 200 OK
pre-render components/link/inline 200 OK
pre-render components/link/standalone 200 OK
pre-render components/modal 200 OK
pre-render components/stepper 200 OK
pre-render components/table 200 OK
pre-render components/tabs 200 OK
pre-render components/tag 200 OK
pre-render components/toast 200 OK
pre-render overrides/power-select 200 OK
pre-render utilities/disclosure 200 OK
pre-render utilities/dismiss-button 200 OK
pre-render utilities/interactive 200 OK
pre-render testing/components/badge 200 OK
pre-render testing/components/banner 200 OK
pre-render testing/components/copy-button 200 OK
pre-render testing/components/do-dont 200 OK
pre-render testing/components/layout 200 OK
pre-render testing/components/typography 200 OK
pre-render testing/markdown/basic-styling 200 OK
pre-render testing/markdown/basic-spacing 200 OK
pre-render testing/markdown/more-complex-code 200 OK
pre-render testing/markdown/more-complex-images 200 OK
pre-render testing/markdown/more-complex-lists 200 OK
pre-render testing/markdown/more-complex-tables 200 OK
pre-render testing/markdown/with-ember-components 200 OK
sitemap.xml successfully created!
cleaning up...
Built project successfully. Stored in "dist/".
File sizes:
 - dist/assets/chunk.143.3f06a5748532934db03d.js: 264.52 kB (30.75 kB gzipped)
 - dist/assets/chunk.178.141b6ee4419bcb746da3.js: 1.6 kB (831 B gzipped)
 - dist/assets/chunk.44.3796bb231f955da24a1a.js: 574.38 kB (145.06 kB gzipped)
 - dist/assets/chunk.556.6e0c45499ee44a6447b3.css: 637 B (286 B gzipped)
 - dist/assets/chunk.556.6e0c45499ee44a6447b3.js: 143 B (122 B gzipped)
 - dist/assets/chunk.730.5695566bf9df560eda19.js: 11.62 kB (3.93 kB gzipped)
 - dist/assets/chunk.942.843a1df3361ba4ffcf77.js: 113.97 kB (36.63 kB gzipped)
 - dist/assets/vendor-2e6c0e62446871ea4d68d77b5b3ae326.css: 1.61 kB (657 B gzipped)
 - dist/assets/vendor-d48bb327a3d2070dbb64c4213e93d263.js: 1.17 MB (290.67 kB gzipped)
 - dist/assets/website-6f4a7a8718251698c15d6d65df51dbef.css: 203.25 kB (27.89 kB gzipped)
 - dist/assets/website-9b040af5941fd6af6f152bf8f0ab86cb.js: 263.38 kB (44.78 kB gzipped)
 - dist/assets/website-fastboot-f09cc45e29a900d319d0d06379e7e9bb.js: 1.9 kB (837 B gzipped)
 - dist/ember-fetch/fetch-fastboot-e28194c4f1288a25de407aad6f7fab09.js: 887 B (510 B gzipped)
 ```
 so clearly the "production" build is not taking into account the filtered list of URLs, somehow? Any idea why and how we could solve it?

Reply:
The sitemap doesn't show the hidden pages: https://hds-website-git-new-docs-website-frontmatter-updates-hashicorp.vercel.app/sitemap.xml

Reply:
🤦  as discussed, I was not clear enough in my review and this is actually working